src/app/child-dev-project/notes/model/note.ts

```diff
@@ -15,26 +15,44 @@
  *     along with ndb-core.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { MeetingNoteAttendance } from "../meeting-note-attendance";
 import { DatabaseEntity } from "../../../core/entity/database-entity.decorator";
 import { Entity } from "../../../core/entity/entity";
 import { DatabaseField } from "../../../core/entity/database-field.decorator";
 import { WarningLevel, WarningLevelColor } from "../../warning-level";
 import { InteractionType } from "../note-config-loader/note-config.interface";
+import { EventAttendance } from "../../attendance/model/event-attendance";
+import { AttendanceStatus } from "../../attendance/model/attendance-status";
+import { v4 as uuid } from "uuid";
 
 @DatabaseEntity("Note")
 export class Note extends Entity {
+  static create(date: Date, subject: string = ""): Note {
+    const instance = new Note(uuid());
+    instance.date = date;
+    instance.subject = subject;
+    return instance;
+  }
+
   /** IDs of Child entities linked with this note */
   @DatabaseField() children: string[] = [];
-  /** An array of triplets containing information about the child and it's attendance */
-  @DatabaseField() attendances: MeetingNoteAttendance[] = [];
+  /** optional additional information about attendance at this event for each of the linked children */
+  @DatabaseField() private childrenAttendance: Map<
+    string,
+    EventAttendance
+  > = new Map();
+
   @DatabaseField() date: Date;
   @DatabaseField() subject: string = "";
   @DatabaseField() text: string = "";
   @DatabaseField() author: string = "";
-  @DatabaseField({ dataType: "interaction-type" }) category: InteractionType = {
-    name: "",
-  };
+  @DatabaseField({ dataType: "interaction-type" }) category: InteractionType =
+    InteractionType.NONE;
+
+  /**
+   * reference to a different entity (e.g. a recurring activity) this note is related to
+   */
+  relatesTo: Entity; // TODO: DatabaseField !! using an entity loader index  id <-> entity instance
```

Comment:
Is there a reason why we cant just put the ID of the activity here? I only see it used when looking for events that belong to a certain activity.

Reply:
I am using the referenced activity object now to check whether the event is assigned to the current user. Want to avoid having to load these from the db separately again.

Maybe the pain is big enough for me to work on #228  in order to resolve this

Reply:
worked around it and found a pretty good solution, I hope.